src/python_testing/TC_EGCTestBase.py

```diff
@@ -0,0 +1,73 @@
+#
+#    Copyright (c) 2025 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+
+import logging
+from datetime import datetime, timedelta, timezone
+
+import chip.clusters as Clusters
+from chip.clusters.Types import NullValue
+from chip.testing import matter_asserts
+from mobly import asserts
+
+logger = logging.getLogger(__name__)
+
+
+class ElectricalGridConditionsTestBaseHelper:
+
+    async def test_ForecastConditions(self,
+                                      endpoint: int = None,
```

Comment:
typing is off: if something can be None it has to be `Optional[...]`.

Why are the defaults none here? it seems we say at least forecastConditions must be a list so it cannot be None. it also seems that the other calls would need their parameters to not be None.

Applies throughout.